lms/djangoapps/courseware/tests/test_discussion_module.py

```diff
@@ -24,3 +26,57 @@ def test_html_with_user(self):
         html = fragment.content
         self.assertIn('data-user-create-comment="false"', html)
         self.assertIn('data-user-create-subcomment="false"', html)
+
+    def test_discussion_render_successfully_with_orphan_parent(self):
+        """
+        Test that discussion module render successfully
+        if discussion module is child of an orphan.
+        """
+        user = UserFactory.create()
+        store = modulestore()
+        course = store.create_course('testX', 'orphan', '123X', user.id)
+        orphan_sequential = store.create_item(self.user.id, course.id, 'sequential')
+
+        vertical = store.create_child(
+            user.id,
+            orphan_sequential.location,
+            'vertical',
+            block_id=course.location.block_id
+        )
+
+        discussion = store.create_child(
+            user.id,
+            vertical.location,
+            'discussion',
+            block_id=course.location.block_id
+        )
+
+        discussion = store.get_item(discussion.location)
+
+        grand_parent = self.get_grand_parent(discussion)
+        # Assert that orphan sequential is last grand parent of discussion module.
+        self.assertEqual(orphan_sequential.location, grand_parent.location)
+
+        discussion = get_module_for_descriptor_internal(
+            user=user,
+            descriptor=discussion,
+            student_data=Mock(name='student_data'),
+            course_id=course.id,
+            track_function=Mock(name='track_function'),
+            xqueue_callback_url_prefix=Mock(name='xqueue_callback_url_prefix'),
+            request_token='request_token',
+        )
+
+        fragment = discussion.render('student_view')
+        html = fragment.content
+        self.assertIn('data-user-create-comment="false"', html)
```

Comment:
Could you do a test to see if the discussion module is an `ErrorModule` or a discussion module? Like in https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/tests/test_error_module.py#L142-L144?